### PR TITLE
Submit expense draft with invitee payee slug

### PR DESCRIPTION
**Pull request by hdiniz**
The mutation allows an AccountReferenceInput, it should handle a payee slug.

### CODE REVIEW

File: server/graphql/common/expenses.ts [hunk 1]
```diff
@@ -131,6 +131,17 @@ const isDraftPayee = async (req: express.Request, expense: Expense): Promise<boo
...
+    const payee = await models.Collective.findOne({
+      attributes: ['id'],
+      where: { slug: expense.data?.payee?.slug },
+    });
+
+    if (!payee) {
+      return false;
+    }
+
+    return req.remoteUser.isAdmin(payee.id);
```
Comment (Betree) [inline server/graphql/common/expenses.ts:144]: Simpler to rather use `fetchAccountWithReference(pick(expense.data?.payee, ['id', 'legacyId', 'slug']))` here?

File: server/graphql/v2/mutation/ExpenseMutations.ts [hunk 2]
```diff
@@ -183,7 +183,8 @@ const expenseMutations = {
       // NOTE(oauth-scope): Ok for non-authenticated users, we only check scope
       enforceScope(req, 'expenses');
 
-      const getId = (reference: { id?: string | number; legacyId?: number }) => reference?.id || reference?.legacyId;
+      const getId = (reference: { id?: string | number; legacyId?: number; slug?: string }) =>
```
Comment (Betree) [inline server/graphql/v2/mutation/ExpenseMutations.ts:186]: :ice_cream: Explicit variable name

```suggestion
      const isExistingAccountReference = (reference: { id?: string | number; legacyId?: number; slug?: string }) =>
```

**Review by Betree — COMMENTED** [review]
Assigning @kewitz for the review since this was implemented in https://github.com/opencollective/opencollective-api/pull/7968.

**Review by kewitz — APPROVED** [review]
Looks good.

**Pull request merged**